### DISCUSSION TITLE
Add cooldownPeriod in HTTPScaledObject

### DISCUSSION THREAD

**Discussion by veptechno** [kedacore/http-add-on#337] [Ideas] [answered]
Right now cooldownPeriod in ScaledObject is not set, so this field has the default value. But in my project, it is important to set this value for each deployment separately.
I think this field would be useful. If you approve, I can try to add it.

**Comment by tomkerkhove** [comment, marked as answer]
@arschles Can we change it so that the default SO "control" flags are available through the HTTPScaledObject?

**Reply by arschles** [reply]
@tomkerkhove yes indeed, we can. @veptechno I'd be happy to review a PR if you'd like to submit one.

**Reply by veptechno** [reply]
@tomkerkhove From what I can see, we need to add only `cooldownPeriod`. It looks like `poolingInterval` value 1s should remain default and there is no need for it to be configurable, and there are no other control flags as I see.

**Reply by arschles** [reply]
@veptechno how about we start with `cooldownPeriod` to see how it looks and avoid blowup of scope. If we do want to add more flags, we can do so in follow-ups.

**Reply by tpiperatgod** [reply]
This is similar to the need in issue #332, so maybe it can be integrated into the discussion?

**Reply by arschles** [reply]
@tpiperatgod do you see a reason to ever have the operator _not_ set `spec.advanced.restoreToOriginalReplicaCount` to true? If we can always set it to true, we don't have to change anything with the `HTTPScaledObject`, which would make that work simpler. If not, I agree that we could combine the two pieces of work. I just think we should avoid building something that allows users to arbitrarily set any `ScaledObject` field from within the `HTTPScaledObject`, at least for now.

**Reply by tpiperatgod** [reply]
Get it. Thx!
